Tests/UnitTests/Fatras/Physics/EnergyLossTests.cpp

```diff
@@ -50,7 +53,25 @@ BOOST_DATA_TEST_CASE(BetheHeitler, Dataset::parameters, pdg, phi, lambda, p,
   BOOST_CHECK_LT(after.absoluteMomentum(), before.absoluteMomentum());
   BOOST_CHECK_LT(after.energy(), before.energy());
   // energy loss creates no new particles
-  BOOST_CHECK(outgoing.empty());
+  BOOST_CHECK_EQUAL(outgoing.size(), 1);
+  BOOST_CHECK_GT(outgoing[0].absoluteMomentum(), 0.);
+  BOOST_CHECK_EQUAL(outgoing[0].pathInX0(), 0.);
+  BOOST_CHECK_EQUAL(outgoing[0].pathInL0(), 0.);
+
+  // Get the four momenta
+  Acts::Vector4 p0 = before.fourMomentum();
+  Acts::Vector4 p1 = after.fourMomentum();
+  Acts::Vector4 k = outgoing[0].fourMomentum();
+
+  // Test for similar invariant masses
+  Acts::Vector4 sum = p1 + k;
+  double s = sum(Acts::eEnergy) * sum(Acts::eEnergy) -
+             sum.template segment<3>(Acts::eMom0).norm() *
+                 sum.template segment<3>(Acts::eMom0).norm();
+  double s0 = p0(Acts::eEnergy) * p0(Acts::eEnergy) -
+              p0.template segment<3>(Acts::eMom0).norm() *
+                  p0.template segment<3>(Acts::eMom0).norm();
+  CHECK_CLOSE_OR_SMALL(s, s0, 1e-2, 1e-2);
```

Comment:
The comparison should probably be smaller and closer to the floating point epsilon. There is no approximations here that would justify requiring on 1% precision on the four-momentum conversion, right? This should be exact within the floating point error.

Reply:
Physically you have an interaction with the material. So it should be just approximately the same.
Considering the calculations done here, those two should be smaller, indeed. However, there are a few cases in which you find small deviations in `sum` compared to the exact solution. As `s0` is of the order of `1e-7`, any numerical inaccuracy would lead to an error for those cases in this test. I assume that this happens due to the normalisation of the direction vector.

Reply:
If you have approximations, e.g. only leading order terms instead of full sin/cos functions, then your method can never reproduce the numerically correct value. In that case, the `CHECK_CLOSE_OR_SMALL` test verifies that the approximations are good enough and the 1% tolerance would be reasonable.

But that is not the case here, right? There are no approximations and all deviations from the reference value originate from floating point inaccuracies only. In such a case, the tolerance should be closer to the floating point epsilon. Since you are comparing quadratic computations here, something like `2-4 * eps * eps` would be a better tolerance. Otherwise, your checks are too loose.

If you see that such a value leads to issues (as you mentioned) than we should check how it can be fixed. But this is probably beyond the scope of this PR.

This is not